### PR TITLE
fix: Admin UI for updated site branch configs and domains

### LINKED ISSUE #4236: Review, identify, and fix any link issues in the Admin UI with site, branch config, and domain model restructuring
An admin user expects all of a site's custom domains, branch configs, and build view links work as expected.

## Acceptance Criteria

- [x] Update site's domains section to pull from the custom domains
- [x] Fix the build view links to point to the preview, demo, live, or custom domain of the build
- [x] Add the related site branch configs to the site

### PR DESCRIPTION
**Pull request by apburnes**
Closes #4236

## Changes proposed in this pull request:
- Update admin's site page to show the associated domains and their state
- Update admin's site page to show the associated site branch configs
- Update admin's site page to show site links to all domains based on site branch context

## security considerations
None

### CODE REVIEW

File: admin-client/src/components/SiteMetadata.svelte [hunk 1]
```diff
@@ -55,8 +55,16 @@
...
+        {#each siteLinks as siteLink, index}
+          <ExternalLink
+            href={siteLink.url}
+            className={index > 0 ? 'padding-left-1' : ''}
+          >
+            {siteLink.context}
+          </ExternalLink>
+        {/each}
+      </div>
+      <div>
```
Comment (drewbo) [inline admin-client/src/components/SiteMetadata.svelte:67]: The extra `div` here puts the repo link on a new line which ends up a little squished. Probably fine to have the new line if we add a little padding.

Comment (drewbo) [reply admin-client/src/components/SiteMetadata.svelte:67]: Also I can't see why but the underline continues past the word on each link? Not essential to fix unless you see something obvious

**Review by drewbo — APPROVED** [review]
Looks good, two small notes but fine to merge prior

File: admin-client/src/components/SiteMetadata.svelte (hunk 1 above)
Comment (apburnes) [reply admin-client/src/components/SiteMetadata.svelte:67]: I'll add some padding to the repo block to add space. I also wasn't able to quickly identify the underscore overhang. I'll take another look with the repo padding fixes.

**Review by drewbo — APPROVED** [review]

**Pull request merged**